backend/lib/src/data/indexer_db/db_client.rs

```diff
@@ -0,0 +1,141 @@
+//! Database client wrapper using repository pattern abstraction
+//!
+//! This module provides a database client that delegates all operations
+//! to an underlying repository implementation, allowing for both production
+//! PostgreSQL and mock implementations for testing.
+
+use std::sync::Arc;
+
+use shc_indexer_db::models::Bsp;
+
+use crate::data::indexer_db::repository::StorageOperations;
+
+/// Database client that delegates to a repository implementation
+///
+/// This client provides a clean abstraction over database operations,
+/// delegating all actual work to an underlying repository that implements
+/// the `StorageOperations` trait. This allows for easy swapping between
+/// production PostgreSQL and mock implementations for testing.
+///
+/// ## Usage Example
+/// ```ignore
+/// use repository::{Repository, StorageOperations};
+/// use data::postgres::DBClient;
+///
+/// // Production usage with PostgreSQL
+/// let repo = Repository::new(&database_url).await?;
+/// let client = DBClient::new(Arc::new(repo));
+///
+/// // Test usage with mock (when available)
+/// let mock_repo = MockRepository::new();
+/// let client = DBClient::new(Arc::new(mock_repo));
+/// ```
+#[derive(Clone)]
+pub struct DBClient {
+    repository: Arc<dyn StorageOperations>,
+}
+
+impl DBClient {
+    /// Create a new database client with the given repository
+    ///
+    /// # Arguments
+    /// * `repository` - Repository implementation to use for database operations
+    pub fn new(repository: Arc<dyn StorageOperations>) -> Self {
+        Self { repository }
+    }
+
+    /// Test the database connection
+    pub async fn test_connection(&self) -> crate::error::Result<()> {
+        // Try to list BSPs with a limit of 1 to test the connection
+        self.repository
+            .list_bsps(1, 0)
+            .await
+            .map_err(|e| crate::error::Error::Database(e.to_string()))?;
+        Ok(())
+    }
+
+    /// Get all BSPs with optional pagination
+    pub async fn get_all_bsps(
+        &self,
+        limit: Option<i64>,
+        offset: Option<i64>,
+    ) -> crate::error::Result<Vec<Bsp>> {
+        let limit = limit.unwrap_or(100);
```

Comment:
Is there any constant to change this? I think that I saw somewhere the limit from the response 🤔

Reply:
No, but there is now